agent/osv_agent.py

```diff
@@ -30,36 +30,38 @@ def process(self, message: m.Message) -> None:
         """Process messages of type v3.asset.file and scan dependencies against vulnerabilities.
         Once the scan is completed, it emits messages of type : `v3.report.vulnerability`
         """
-        self._osv_file_handler: osv_file_handler.OSVFileHandler
         logger.info("processing message of selector : %s", message.selector)
         content = message.data.get("content")
         path = message.data.get("path")
+        if path is None:
+            logger.error("Can't process empty path.")
+            return
         if content is None or content == b"":
             logger.warning("Message file content is empty.")
             return
-        self._osv_file_handler = osv_file_handler.OSVFileHandler(
-            content=content, path=path
-        )
 
-        self._run_osv(content)
+        self._run_osv(content, path)
 
-    def _run_osv(self, content: bytes) -> None:
+    def _run_osv(self, content: bytes, path: str) -> None:
         """Perform the osv scan with two flags with --sbom and --lockfile,  letting OSV validate the file
          instead of guessing the file format.
         Args:
             content: Scanned file content
+
+            path: The file path.
         """
-        extension = self._osv_file_handler.get_file_type()
+        file_name = pathlib.Path(path).name
```

Comment:
If you use pathlib, don't use open. Use its own primitives.

Reply:
If I want to use the pathlib primitive i need to convert the  last part of the file to a path object, which is less elegant.